packages/dx-react-grid/docs/reference/column-chooser.md

```diff
@@ -0,0 +1,38 @@
+# Grid Reference
+
+ColumnChooser is a component designed to work along with Grid. It displays all available columns and in a combination with the [TableColumnVisibility](table-column-visibility.md) Grid plugin enables an end-user to control which columns should be displayed.
+
+## User reference
+
+### Properties
+
+Name | Type | Default | Description
+-----|------|---------|------------
+columns | Array&lt;[Column](grid.md#column)&gt; | | Specifies for which row object fields columns are created.
+hiddenColumns | Array&lt;string&gt; | [] | An array containing the names of the columns to be hidden.
+onHiddenColumnsChange | (nextHiddenColumns: Array&lt;string&gt;) => void | | Handles column visibility change.
+contentTemplate | (args: [ColumnChooserRootArgs](#column-chooser-root-args)) => ReactElement | | A template that renders the column chooser markup.
```

Comment:
contentTemplate with ColumnChooserRootArgs... It's quite strange

Reply:
Also seems that we should extract container and item templates for API consistency